src/Core/Registration/ColoredICP.cpp

```diff
@@ -63,6 +65,10 @@ class TransformationEstimationForColoredICP : public TransformationEstimation {
 
 public:
     double lambda_geometric_;
+
+private:
+    const TransformationEstimationType type_ =
+            TransformationEstimationType::ColoredICP;
```

Comment:
I am not very sure whether this private variable is necessary for every child class of TransformationEstimation. The same concern applies to TransformationEstimationPointToPoint and TransformationEstimationPointToPlane.

Reply:
I think this is fine. Usually, it is not recommended to let the subclasses keep the knowledge of which class it is. But here we don't have other options (or we have to let ComputeEstimation() yell every time it gets called).

And we do this to Geometry class too. Every subclass has a type_ which indicates which type it is. So I think it is fine.

The downside though, is that it will be hard to extend - say, someone wants to write a customized TransformationEstimation using Python. Then there might be other problems. Anyways, we will worry about it later.

Reply:
In fact, we do want to write a customized Python subclass of the TransformationEstimation class. For example, 
"class TransformationEsimationTranslation(TransformationEstimation):
      def ...
"
I wonder if we can do that in Python, or we also have to change the pybind11 interfaces? Thanks!

Reply:
I did it before in C++. It is doable.

However I don't think it is doable in Python in its current way (or at least I have not tested).

There are performance consideration though. These functions are called extensively and you definitely want them to be written in native C++ in production code.

Reply:
It makes sense to do it in C++. I wonder if it is straightforward for developers to change Open3D  C++ code to do translation-only ICP. Or for this particular case, we have to change some of the Eigen functions, like the Umeyama implementation. Thanks!